walk-in-interview/src/main/webapp/account/business/edit/script.js

```diff
@@ -0,0 +1,156 @@
+  const queryString = window.location.search;
+
+  const urlParams = new URLSearchParams(queryString);
+  if (urlParams === '') {
+    throw new Error('url params should not be empty');
+  }
+
+  return urlParams.get(BUSINESS_ID_PARAM);
+}
+
+/**
+ * Gets the business details from the url.
+ * @return {JSON} The business detail json.
+ */
+function getBusinessDetails() {
+  const queryString = window.location.search;
+
+  const urlParams = new URLSearchParams(queryString);
+  if (urlParams === '') {
+    throw new Error('url params should not be empty');
+  }
+
+  return JSON.parse(urlParams.get(BUSINESS_DETAILS_PARAM));
+}
+
+function getBusinessDetailsFromUserInput() {
+  const businessId = getBusinessId();
+  const businessName = document.getElementById('name').value;
+
+  const jobsList = getBusinessDetails().jobs;
+
+  const businessDetails = {
+    accountId: businessId,
+    name: businessName,
+    jobs: jobsList,
+  };
+
+  return businessDetails;
+}
+
+/**
+* Validates the user input.
+* Shows error message on the webpage if there is field with invalid input.
+*
+* @return {boolean} depending on whether the input is valid or not.
+*/
+function validateRequiredUserInput() {
+  const businessId = getBusinessId();
+  const name = document.getElementById('name').value;
+
+  if (businessId === '') {
+    setErrorMessage(/* errorMessageElementId= */'error-message',
+        /* msg= */ 'Empty Business Id found.',
+        /* includesDefault= */false);
+    return false;
+  }
+
+  if (name === '') {
+    setErrorMessage(/* errorMessageElementId= */'error-message',
+        /* msg= */ STRINGS['name']);
+    return false;
+  }
+
+  return true;
+}
+
+const backButton = document.getElementById('back');
+backButton.addEventListener('click', (_) => {
+  window.location.href = HOMEPAGE_PATH;
+});
+
+const submitButton = document.getElementById('submit');
+submitButton.addEventListener('click', (_) => {
+  if (!validateRequiredUserInput()) {
+    return;
+  }
+
+  const accountDetails = getBusinessDetailsFromUserInput();
+
+  fetch(API['update-business-account'], {
+    method: 'PATCH',
+    headers: {'Content-Type': 'application/json'},
+    body: JSON.stringify(accountDetails),
+  })
+      .then((response) => {
+        if (response.status == BAD_REQUEST_STATUS_CODE) {
+          setErrorMessage(/* errorMessageElementId= */'error-message',
+              /* msg= */ STRINGS['update-error-message'],
+              /* includesDefault= */false);
+          throw new Error(STRINGS['update-error-message']);
+        }
+
+        /** reset the error (there might have been an error msg from earlier) */
+        setErrorMessage(/* errorMessageElementId= */'error-message',
+            /* msg= */ '', /* includesDefault= */false);
+        window.location.href = HOMEPAGE_PATH;
+      })
+      .catch((error) => {
+        // Not the server response error already caught and thrown
+        if (error.message != STRINGS['update-error-message']) {
```

Comment:
Maybe better if we have different catch for different type of error instead of comparing error message?